src/cbc_sdk/enterprise_edr/auth_events.py

```diff
@@ -0,0 +1,690 @@
+            max_events_per_group (int): Maximum number of events in a group, if not provided all events will be returned
+            rows (int): Number of rows to request, can be paginated
+            start (int): First row to use for pagination
+            ranges (dict): dict with information about duration, field, method
+
+        Returns:
+            dict: grouped results
+
+        Examples:
+            >>> cb = CBCloudAPI(profile="example_profile")
+            >>> groups = set(cb.select(AuthEvents).where(process_pid=2000).group_results("device_name"))
+            >>> for group in groups:
+            >>>     print(group._info)
+        """
+        if not isinstance(fields, list) and not isinstance(fields, str):
+            raise ApiError("Fields should be either a single field or list of fields")
+
+        if isinstance(fields, str):
+            fields = [fields]
+
+        if not all((gf in AuthEventsQuery.VALID_GROUP_FIELDS) for gf in fields):
+            raise ApiError("One or more invalid aggregation fields")
+
+        if not self._query_token:
+            self._submit()
+
+        result_url = "/api/investigate/v2/orgs/{}/auth_events/search_jobs/{}/group_results".format(
+            self._cb.credentials.org_key,
+            self._query_token,
+        )
+
+        # construct the group results body, required ones are fields and rows
+        data = dict(fields=fields, rows=rows)
+        if max_events_per_group is not None:
+            data["max_events_per_group"] = max_events_per_group
+        if range_duration or range_field or range_method:
+            data["range"] = {}
+            if range_method:
+                data["range"]["method"] = range_method
+            if range_duration:
+                data["range"]["duration"] = range_duration
+            if range_field:
+                data["range"]["field"] = range_field
+        if start is not None:
+            data["start"] = start
+
+        still_fetching = True
+        while still_fetching:
+            result = self._cb.post_object(result_url, data).json()
+            contacted = result.get("contacted", 0)
+            completed = result.get("completed", 0)
+            if contacted < completed:
+                time.sleep(0.5)
+                continue
+            still_fetching = False
+
+        for group in result.get("group_results", []):
+            yield AuthEventsGroup(self._cb, initial_data=group)
+
+    def get_auth_events_descriptions(self):
+        """
+        Returns descriptions and status messages of Auth Events.
+
+        Returns:
+            dict: Descriptions and status messages of Auth Events as dict objects.
+        Example:
+            >>> cb = CBCloudAPI(profile="example_profile")
+            >>> descriptions = cb.select(AuthEvents).get_auth_events_descriptions()
+            >>> print(descriptions)
+        """
+        url = "/api/investigate/v2/orgs/{}/auth_events/descriptions".format(self._cb.credentials.org_key)
```

Comment:
Should this be added to the CBCloudAPI instead? Feels a little weird on a query object since we don't create query objects directly.

Reply:
It felt off to be in the CBCloudAPI for UX purposes mainly;
It will not show up in the RTD on the Auth Events page, so an API user familiar with the functionality will be confused about how to get the event descriptions. A new user will probably miss the functionality in general. 

We would need to add additional notes and guides, possibly to tell the user this exists.
Personally, as a (lazy)user, I usually do this instead of reading docs:
`
>>> cb = CBCloudAPI(profile="")
>>> query = cb.select(AuthEvents)
>>> dir(query)
`['VALID_GROUP_FIELDS', '__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__iter__', '__le__', '__len__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_add_exclusions', '_batch_size', '_cb', '_clone', '_count', '_count_valid', '_criteria', '_default_args', '_doc_class', '_exclusions', '_fields', '_get_query_parameters', '_group_by', '_init_async_query', '_perform_query', '_query', '_query_builder', '_query_token', '_run_async_query', '_search', '_sort_by', '_start', '_still_querying', '_submit', '_time_range', '_timed_out', '_timeout', '_total_results', '_update_criteria', '_validate', 'add_criteria', 'add_exclusions', 'all', 'and_', 'batch_size', 'execute_async', 'first', 'get_auth_events_descriptions', 'group_results', 'not_', 'one', 'or_', 'search_suggestions', 'search_validation', 'set_fields', 'set_rows', 'set_start', 'set_time_range', 'sort_by', 'timeout', 'update_criteria', 'where']

To me, this feels like it should be a part of the AuthEvents object, but if CBCloudAPI is our approach, I'm happy to move it there.

Reply:
Fixed as per our SDK discussions yesterday.